x-pack/plugins/ml/public/privilege/get_privileges.js

```diff
@@ -64,94 +65,20 @@ export function getPrivileges() {
-            resp.cluster['cluster:monitor/xpack/ml/datafeeds/stats/get']) {
-            privileges.canGetDatafeeds = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/job/put'] &&
-            resp.cluster['cluster:admin/xpack/ml/job/open'] &&
-            resp.cluster['cluster:admin/xpack/ml/datafeeds/put']) {
-            privileges.canCreateJob = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/job/update']) {
-            privileges.canUpdateJob = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/job/open']) {
-            privileges.canOpenJob = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/job/close']) {
-            privileges.canCloseJob = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/job/forecast']) {
-            privileges.canForecastJob = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/job/delete'] &&
-            resp.cluster['cluster:admin/xpack/ml/datafeeds/delete']) {
-            privileges.canDeleteJob = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/job/open'] &&
-            resp.cluster['cluster:admin/xpack/ml/datafeeds/start'] &&
-            resp.cluster['cluster:admin/xpack/ml/datafeeds/stop']) {
-            privileges.canStartStopDatafeed = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/datafeeds/update']) {
-            privileges.canUpdateDatafeed = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/datafeeds/preview']) {
-            privileges.canPreviewDatafeed = true;
-          }
-
-          if (resp.cluster['cluster:monitor/xpack/ml/calendars/get']) {
-            privileges.canGetCalendars = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/calendars/put'] &&
-            resp.cluster['cluster:admin/xpack/ml/calendars/jobs/update'] &&
-            resp.cluster['cluster:admin/xpack/ml/calendars/events/post']) {
-            privileges.canCreateCalendar = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/calendars/delete'] &&
-            resp.cluster['cluster:admin/xpack/ml/calendars/events/delete']) {
-            privileges.canDeleteCalendar = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/filters/get']) {
-            privileges.canGetFilters = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/filters/put'] &&
-            resp.cluster['cluster:admin/xpack/ml/filters/update']) {
-            privileges.canCreateFilter = true;
-          }
-
-          if (resp.cluster['cluster:admin/xpack/ml/filters/delete']) {
-            privileges.canDeleteFilter = true;
-          }
-
-          if (resp.cluster['cluster:monitor/xpack/ml/findfilestructure']) {
-            privileges.canFindFileStructure = true;
-          }
-
+          setGettingPrivileges(resp.cluster, privileges);
```

Comment:
Suggestion: Personally I'd think it would be more clear what these two functions are doing if they'd return `privileges` again, for example:

```javascript
privileges = applyGettingPrivileges(resp.cluster, privileges);
privileges = applyActionPrivileges(resp.custer, privileges);
```

or even

```javascript
resolve({
  ...applyGettingPrivileges(resp.cluster, privileges),
  ...applyActionPrivileges(resp.custer, privileges),
});
```

Reply:
making the change `privileges = applyGettingPrivileges(resp.cluster, privileges);` would require `privileges` to be changed to a `let`.
and changing this to use a spread operator would require quite a bit of restructuring of the code.
I'm going to leave this as is for now and we can revisit at a later date if needed